File: Problems/FixTheCode/FTC_2_sorting-algorithms_expanded/java/source/PerformanceTest.java
import java.time.Duration;
import java.time.Instant;
import java.util.Arrays;

public class PerformanceTest {
    public void PerformanceTest(){}

    public void EvaluatePerformance(Sorter sorter, int[] array, int[] expectedSortedArray){
        Instant startTime = Instant.now();
        int[] sortedArray =sorter.Sort(array);
        Instant endTime = Instant.now();
        long duration = Duration.between(startTime, endTime).toMillis();

        if (!Arrays.equals(sortedArray, expectedSortedArray)){
            System.out.println("[ERROR] Sorting failed. Expected: " + java.util.Arrays.toString(expectedSortedArray) + " but got: " + Arrays.toString(sortedArray));
        }
        System.out.println("Sorting took " + duration + " milliseconds.");
    }



}

File: Problems/FixTheCode/FTC_3_sorting-algorithms/python/source/performance_test.py
import time

class PerformanceTest:
    def __init__(self):
        pass

    def evaluate_performance(self, sorter, array, expected_sorted_array):
        start_time = time.time()
        sorter.sort()
        sorted_array = sorter.array
        end_time = time.time()
        duration = (end_time - start_time) * 1000  # Convert to milliseconds

        if sorted_array != expected_sorted_array:
            print("[ERROR] Sorting failed. Expected array different from sorted one.")
        print(f"Sorting took {duration:.2f} milliseconds.")